using-mobile-extensions/adobe-experience-platform-messaging/messaging-api-reference.md

```diff
@@ -0,0 +1,420 @@
+After retrieving the push token use the below core api to sync it with profile in platform.
+
+### Java
+
+#### Syntax
+
+```java
+public static void setPushIdentifier(final String pushIdentifier);
+```
+
+* *pushIdentifier* - A `String` value denoting the push token.
+
+#### Examples
+
+```java
+FirebaseMessaging.getInstance().getToken()
+        .addOnCompleteListener(new OnCompleteListener<String>() {
+            @Override
+            public void onComplete(@NonNull Task<String> task) {
+                if (task.isSuccessful()) {
+                    String token = task.getResult();
+                    MobileCore.setPushIdentifier(token);
+                }
+            }
+        });
+```
+
+{% endtab %}
+
+{% tab title="iOS" %}
+
+### Swift
+
+To retrieve the push token in iOS, checkout the apple documentation [here](https://developer.apple.com/documentation/usernotifications/registering_your_app_with_apns).  
+After retrieving the push token use the below core api to sync it with profile in platform.
+
+#### Syntax
+
+```swift
+public static func setPushIdentifier(_ deviceToken: Data?)
+```
+
+* *deviceToken* - A `String` value denoting the push token..
+
+#### Examples
+
+```swift
+func application(_: UIApplication, didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data) {
+    MobileCore.setPushIdentifier(deviceToken)
+}
+```
+
+### Objective-C
+
+#### Syntax
+
+```objective-c
+public static func setPushIdentifier(_ deviceToken: Data?)
+```
+
+#### Examples
+
+```objective-c
+- (void)application:(UIApplication *)application didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceToken { 
+    [AEPMobileCore setPushIdentifier:deviceToken];
+}
+```
+
+{% endtab %}
+{% endtabs %}
+
+## addPushTrackingDetails<a id="addpushtrackingdetails"></a>
+
+This API is used to updated the push notification pending intent with necessary customer journey information.
+{% hint style="warning" %}
+Call to this API is necessary to ensure that all the important information (messageId, Customer journey information) are added to the pending intent so that they can be used while tracking the push notification interactions.
+{% endhint %}
+
+{% tabs %}
+{% tab title="Android" %}
+
+### Java
+
+#### Syntax
+
+```java
+public static boolean addPushTrackingDetails(final Intent intent, final String messageId, final Map<String, String> data)
+```
+
+* *intent* - `Intent` which is added to the pending intent so that it can be used when user interacts with the notification.
+* *messageId* - `String` message ID of the push notification
+* *data* - `Map` which represents the data part of the remoteMessage.
+* Returns `boolean` indicating whether the intent was update with necessary information (messageId and customer journey data).
+
+
+#### Examples
+
+```java
+boolean update = addPushTrackingDetails(intent, messageId, data)
```

Comment:
should name the var something like `success`.

it's also unclear to me when the update would fail.  when would calling this method ever return `false`?

Reply:
if messageId is null or something